File: backend/app/schemas/weather.py
from pydantic import BaseModel


class WeatherForecastEntry(BaseModel):
    forecast_date: str
    forecast_time: str
    temperature_celsius: float | None = None
    sky: str | None = None
    precipitation_type: str | None = None
    precipitation_probability: int | None = None
    precipitation_amount: str | None = None
    humidity: int | None = None
    wind_speed: float | None = None


class WeatherForecastResponse(BaseModel):
    base_date: str
    base_time: str
    nx: int
    ny: int
    current: WeatherForecastEntry | None
    forecasts: list[WeatherForecastEntry]

File: frontend/src/types/weather.ts
export interface WeatherForecast {
  forecast_date: string;
  forecast_time: string;
  temperature_celsius: number | null;
  sky: string | null;
  precipitation_type: string | null;
  precipitation_probability: number | null;
  precipitation_amount: string | null;
  humidity: number | null;
  wind_speed: number | null;
}

export interface WeatherCurrent {
  forecast_date: string;
  forecast_time: string;
  temperature_celsius: number | null;
  sky: string | null;
  precipitation_type: string | null;
  precipitation_probability: number | null;
  precipitation_amount: string | null;
  humidity: number | null;
  wind_speed: number | null;
}

export interface WeatherResponse {
  base_date: string;
  base_time: string;
  nx: number;
  ny: number;
  current: WeatherCurrent | null;
  forecasts: WeatherForecast[];
}
